servlet-core/src/main/java/io/micronaut/servlet/http/ServletHttpHandler.java

```diff
@@ -396,50 +413,50 @@ private void subscribeToResponsePublisher(HttpRequest<Object> req,
                                         encodeResponse(exchange, annotationMetadata, response);
                                         return res1;
                                     }
-                                }).switchIfEmpty(Flowable.defer(() -> {
+                                }).switchIfEmpty(Flux.defer(() -> {
                                     final RouteMatch<Object> errorRoute = lookupStatusRoute(route, HttpStatus.NOT_FOUND);
                                     if (errorRoute != null) {
-                                        Flowable<MutableHttpResponse<?>> notFoundFlowable = Flowable.fromPublisher(buildResponsePublisher(
+                                        Flux<MutableHttpResponse<?>> notFoundFlux = Flux.from(buildResponsePublisher(
                                                 req,
                                                 (MutableHttpResponse<Object>) response,
                                                 errorRoute,
                                                 true
                                         ));
-                                        return notFoundFlowable.onErrorReturn(throwable -> {
+                                        return notFoundFlux.onErrorResume(throwable -> {
                                             if (LOG.isErrorEnabled()) {
-                                                LOG.error("Error occuring invoking 404 handler: " + throwable.getMessage());
+                                                LOG.error("Error occurring invoking 404 handler: " + throwable.getMessage());
```

Comment:
This should use parameterised logging instead. 

Instead of 

`LOG.error("Error occurring invoking 404 handler: " + throwable.getMessage());`

It should be 

`LOG.error("Error occurring invoking 404 handler: {}", throwable.getMessage());`